definitions/checks/foreman_installer/installer_noop_run.rb

```diff
@@ -0,0 +1,22 @@
+module Checks
+  module ForemanInstaller
+    class InstallerNoopRun < ForemanMaintain::Check
+      metadata do
+        description 'Run the installer in noop mode for system checks'
```

Comment:
This will do a full `--noop` run of the installer, which can take some time.

I wonder if the installer should have a `--checks-only` option? cc @ehelms @ekohl

Reply:
Yeah that would be nice!

Reply:
Just the checks implies running `run-parts /usr/share/foreman-installer/checks/*`, but we've implemented the system requirements via a hook. That's tricky since you'll also run all other hooks.

Looking at https://github.com/theforeman/kafo/blob/master/doc/kafo_run.png we can run everything up to (and including) pre_commit. I'm not sure how you'd name it though.

The easiest is some hook in the installer (like `hooks/pre/00-no-real-run.rb` or `hooks/pre_commit/99-no-real-run.rb`) that has a certain condition that makes it quit. Right now I'm thinking of some environment variable. That way you can run with `--noop` AND the env var. If it's supported, it'll be faster and if it isn't, no harm is done since it's still a noop run.

Thoughts?

Reply:
I think implementing one of these strategies is the right way forward. For this PR, I would prefer:

 1) drop the use of noop here and merge this
 2) let us add a feature to the installer
 3) add a fix later that uses the installer's feature

Reply:
> drop the use of noop here and merge this

With this you mean drop the entire installer run and let it fail when it actually runs? I'd be OK with that, but I can't see all the implications in foreman-maintain.

Reply:
Tracking this need here -- https://projects.theforeman.org/issues/35555